integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/GenericFieldIT.java

```diff
@@ -187,6 +193,85 @@ public void customBridge_explicitFieldType() {
 		backendMock.verifyExpectationsMet();
 	}
 
+	@Test
+	public void customBridge_withParameters_annotationMapping() {
+		@Indexed(index = INDEX_NAME)
+		class IndexedEntity {
+			@DocumentId
+			Integer id;
+			@GenericField(valueBinder = @ValueBinderRef(type = ParametricBridge.ParametricBinder.class,
+					params = @Parameter(name = "stringBase", value = "4")))
+			Integer value;
+
+			IndexedEntity() {
+			}
+			IndexedEntity(Integer id, Integer value) {
+				this.id = id;
+				this.value = value;
+			}
+		}
+
+		backendMock.expectSchema( INDEX_NAME, b -> b.field( "value", String.class ) );
+		SearchMapping mapping = setupHelper.start()
+				.expectCustomBeans().setup( IndexedEntity.class );
+		backendMock.verifyExpectationsMet();
+
+		try ( SearchSession session = mapping.createSession() ) {
+			SearchIndexingPlan plan = session.indexingPlan();
+			plan.add( new IndexedEntity( 1, 93 ) );
+			plan.add( new IndexedEntity( 2, null ) );
+
+			backendMock.expectWorks( INDEX_NAME )
+					.add( "1", b -> b.field( "value", "1" ) )
+					.add( "2", b -> b.field( "value", "4" ) )
+					.createdThenExecuted();
+		}
+		backendMock.verifyExpectationsMet();
+	}
+
+	@Test
+	public void customBridge_withParameters_programmaticMapping() {
+		class IndexedEntity {
+			Integer id;
+			Integer value;
+
+			IndexedEntity() {
+			}
+			IndexedEntity(Integer id, Integer value) {
+				this.id = id;
+				this.value = value;
+			}
+		}
+
+		backendMock.expectSchema( INDEX_NAME, b -> b.field( "value", String.class ) );
+		SearchMapping mapping = setupHelper.start()
+				.withConfiguration( builder -> {
+					builder.addEntityType( IndexedEntity.class );
+
+					TypeMappingStep indexedEntity = builder.programmaticMapping().type( IndexedEntity.class );
+					indexedEntity.indexed().index( INDEX_NAME );
+					indexedEntity.property( "id" ).documentId();
+					indexedEntity.property( "value" ).genericField().valueBinder(
+							new ParametricBridge.ParametricBinder(),
+							Collections.singletonMap( "base", 4 )
+					);
+				} )
+				.expectCustomBeans().setup( IndexedEntity.class );
+		backendMock.verifyExpectationsMet();
+
+		try ( SearchSession session = mapping.createSession() ) {
+			SearchIndexingPlan plan = session.indexingPlan();
+			plan.add( new IndexedEntity( 1, 93 ) );
+			plan.add( new IndexedEntity( 2, null ) );
+
+			backendMock.expectWorks( INDEX_NAME )
+					.add( "1", b -> b.field( "value", "1" ) )
+					.add( "2", b -> b.field( "value", "4" ) )
+					.createdThenExecuted();
+		}
+		backendMock.verifyExpectationsMet();
+	}
```

Comment:
Please also test edge cases (see same comment on `DocumentIdBaseIT`).

Not sure it's necessary to duplicate the tests of edge cases for other field annotations, though: that seems overkill.